crates/cairo-lint-core/tests/test_files/equatable_if_let/equatable_if_let

```diff
@@ -0,0 +1,281 @@
+//! > simple equality simple pattern
```

Comment:
what happens in that case:
```rs
fn main() {
    let x = Option::Some(1_felt252);
    if let Option::Some(val) = x {
        println!("{val}");
    }
}
```

Reply:
Hi @0xLucqs , I was analyzing this code. I think that the equatable_if_let lint doesn't trigger because it is destructuring the variable. To you that is OK?

Reply:
yes that's perfect

Reply:
can you add the test plz